src/main/java/org/broadinstitute/hellbender/utils/pileup/PileupElement.java

```diff
@@ -4,11 +4,16 @@
 import htsjdk.samtools.CigarOperator;
 import org.broadinstitute.hellbender.utils.BaseUtils;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.locusiterator.AlignmentStateMachine;
 import org.broadinstitute.hellbender.utils.read.ReadUtils;
 
 import java.util.*;
 
-public final class PileupElement implements Comparable<PileupElement> {
+/**
+ * Represents a position in the read pileup.
+ */
+public final class PileupElement {
+    //FIXME: this static linked list is modifiable
```

Comment:
to clarify - it does not need to be modifiable but there's no easy way to make a nonmodifiable LinkedList. The current code is terrible, I agree. Gift from GATK3.